### ISSUE TITLE
send-error message not always sent / wrong order

### ISSUE THREAD

**Issue by lgrahl** [saltyrtc/saltyrtc-server-python#77]
When a relay message is being enqueued (as a task from id `0x01`) and the subsequent ping (to `0x02`) times out, the resulting `PingTimeoutError` (on a task of `0x02`) for some reason prevents the `send-error` routine from being called.

The relay send task on `0x02` is cancelled and the relay task waiting for the send task to complete on `0x01` doesn't get that information.

**Comment by lgrahl** [comment]
Okay, so I pinned it down: The problem is that we cannot determine which message has been lost, or in fact whether a message has been lost at all. As an example:

1. The initiator sends a relay message to the server.
2. The connection from the server to the responder has already been lost but the server doesn't know yet.
3. The server relays the message to the responder, i.e. it calls `send` on the socket. And that `send` call succeeds because, again, the server doesn't know yet that the connection has been lost.
4. After a while, the server gets a TCP timeout (or a ping timeout) for the responder. But it has no idea whether messages have been lost or not.
5. Since the original `send` call succeeded, it only sends a `disconnected` message to the initiator.

To summarise: We do not know where the stream has been cut off. Thus we cannot (and this is very important to grasp) reliably tell that no data has been lost unless the connection to the other peer has been closed gracefully. Because of this, we may need to re-evaluate the usefulness of `send-error`. A simple flag in the `disconnected` message telling the peer how the connection has been closed would suffice.

/cc @dbrgn

**Comment by lgrahl** [comment]
This issue is still open since I still need to fix that a `send-error` can be sent after a `disconnected` message.
